src/components/logservice/LogServiceButton.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { LogServiceButtonData, FormPropsData } from "./data"
+import { useDisclosure } from '@mantine/hooks';
+import { Modal, Button, Group } from '@mantine/core';
+import { Form } from "./Form";
+
+
+/**
+ * LogServiceButtonProps
+ */
+export type LogServiceButtonProps = LogServiceButtonData & {
+  onSubmit?: (responses: any) => Promise<any>,
+  onClick?: () => void,
+};
+
+/**
+ * LogServiceButton
+ * @param props
+ * @constructor
+ */
+export const LogServiceButton = (props: LogServiceButtonProps) => {
+  const [isOpen, setIsOpen] = React.useState(!!props.isOpen);

```

Comment:
Instead of managing modal state manually, you should use modals manager to do so - https://mantine.dev/others/modals/

Reply:
Per our convo during 1-on-1, in order to use modals developers need to have the modal provider somewhere in the parent tree. An example is available in the codebase as well as in the mantine docs.

```
import { ModalsProvider } from '@mantine/modals';

function Demo() {
  return (
    <ModalsProvider labels={{ confirm: 'Submit', cancel: 'Cancel' }}>
      <App />
    </ModalsProvider>
  );
}
```